src/main/java/graphql/nadel/result/ExecutionResultNode.java

```diff
@@ -24,6 +25,7 @@
     private final List<ExecutionResultNode> children;
     private final List<GraphQLError> errors;
     private final ElapsedTime elapsedTime;
+    private final AtomicInteger totalNodeCount = new AtomicInteger(0);
```

Comment:
We spoke about making Immutable again.  I think we MUSt do it

   `public abstract ExecutionResultNode withNodeCount(int nodeCount);`

and have each class instiate itself with the node count

Reply:
therefore you dont need AtomicInteger since its now an immutable object